### PR TITLE
ARROW-8548: [Website] 0.17 release post

### PR DESCRIPTION
**Pull request by nealrichardson**

### CODE REVIEW

File: _posts/2020-04-21-0.17.0-release.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+The "Feather V2" format based on the Arrow IPC file format was developed.
+Feather v2 features full support for all Arrow data types,
+and resolves the 2GB per-column limitation for large amounts of string data that
+the [original Feather implementation][13] had.
+Feather V2 also introduces experimental IPC message compression using LZ4 frame format or ZSTD. This will be formalized later in the Arrow format.
+
+### C++ Datasets
+
+TODO: SUMMARIZE C++ DATASETS WORK
```
Comment (nealrichardson) [inline _posts/2020-04-21-0.17.0-release.md:70]: @fsaintjacques

File: _posts/2020-04-21-0.17.0-release.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+* Complete support for writing nested types to Parquet format was completed. The legacy code can be accessed through parquet write option C++ and an environment variable in Python. Read support will come in a future release.
+* The BYTE_STREAM_SPLIT encoding was implemented for floating-point types. It helps improve the efficiency of memory compression for high-entropy data.
+* Expose Parquet schema field_id as Arrow field metadata
+* Support for DataPageV2
+
+### C++ build notes
+
+* We continued to make the core C++ library build simpler and faster. Among the improvements are the removal of the dependency on Thrift Compiler at build time; while Parquet still requires `thrift-cpp`, its dependencies are much lighter. We also further reduced the number of build configurations that require `boost`, and when `boost` is needed to be built, we only download the components we need, reducing the size of the `boost` bundle by 90%.
+* TODO: ARM64 platform notes?
+* TODO: LLVM 8 upgrade?
```
Comment (nealrichardson) [inline _posts/2020-04-21-0.17.0-release.md:83]: @kou?

File: _posts/2020-04-21-0.17.0-release.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+* A streaming CSV reader was implemented, yielding individual RecordBatches and helping limit overall memory occupation.
+* Array casting from Decimal128 to integer types and to Decimal128 with different scale/precision was added.
+* Sparse CSF tensors are now supported.
+* When creating an Array, the null bitmap is not kept if the null count is known to be zero
+* Compressor support for the LZ4 frame format (LZ4_FRAME) was added
+* An event-driven interface for reading IPC streams was added.
+* Further core APIs that required passing an explicit out-parameter were migrated to Result<T>.
+
+TODO: New analytics kernels (match, sort indices / argsort, top-k)
+TODO: New Gandiva features?
```
Comment (nealrichardson) [inline _posts/2020-04-21-0.17.0-release.md:100]: @praveenbingo @bkietz

File: _posts/2020-04-21-0.17.0-release.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+* Updated `pyarrow.dataset` module following the changes in the C++ Datasets project. This release also adds [richer documentation][10] on the datasets module.
+* Support for the improved dataset functionality in `pyarrow.parquet.read_table/ParquetDataset`. To enable, pass `use_legacy_dataset=False`. Among other things, this allows to specify filters for all columns and not only the partition keys (using row group statistics) and enables different partitioning schemes. See the "note" in the
+[`ParquetDataset` documentation][11].
+
+### Packaging
+
+* Wheels for Python 3.8 are now available
+* Support for Python 2.7 has been dropped
+* TODO: Summarize where to find out about Nightly Dev Builds
```
Comment (nealrichardson) [inline _posts/2020-04-21-0.17.0-release.md:121]: @kszucs

File: _posts/2020-04-21-0.17.0-release.md (hunk 2 above)
Comment (kou) [reply _posts/2020-04-21-0.17.0-release.md:83]: Will do in a few days.
(If someone can take over this, please do.)

File: _posts/2020-04-21-0.17.0-release.md (hunk 1 above)
Comment (fsaintjacques) [reply _posts/2020-04-21-0.17.0-release.md:70]: * Add the ability to list files in FileSystemDataset
* Add options to ParquetFileFormat
* Support dictionary columns in partition expression
* Relax discovery validation and improve speed on high latency file system
* Support columns projection with IpcFileFormat
* Fix various crashes and other issues

**Comment by pitrou** [comment]
Looks ok to me (except for the TODOs of course).

**Comment by nealrichardson** [comment]
Thanks all for the help. I've resolved the remaining TODOs, so IMO this is ready to merge.

**Pull request closed without merging**